src/features/content-answers/ContentAnswersModalContent.tsx

```diff
@@ -13,28 +13,60 @@ import { User } from '../../../common/types/core';
 import './ContentAnswersModalContent.scss';
 
 type Props = {
-    currentUser: User;
+    currentUser?: User;
     fileName: string;
     isLoading: boolean;
     questions: QuestionType[];
 };
 
 const ContentAnswersModalContent = ({ currentUser, fileName, isLoading, questions }: Props) => {
+    const messagesEndRef = useRef<HTMLDivElement | null>(null);
+
+    const handleScrollToBottom = useCallback(behavior => {
+        if (messagesEndRef.current && messagesEndRef.current.scrollIntoView) {
+            messagesEndRef.current.scrollIntoView({ behavior });
+        }
+    }, []);
+
+    const throttledScrollToBottom = throttle(handleScrollToBottom, 1000);
+
+    // Scroll to the bottom when modal opened
+    useEffect(() => {
+        setTimeout(() => {
+            handleScrollToBottom('instant');
+        }, 0);
+    }, [handleScrollToBottom]);
+
+    // Scroll to the bottom if a new answer is loading
+    useEffect(() => {
+        if (isLoading) {
+            handleScrollToBottom('smooth');
+        }
+    }, [handleScrollToBottom, isLoading]);
+
     return (
-        <div className="bdl-ContentAnswersModalContent">
+        <div className="bdl-ContentAnswersModalContent" data-testid="content-answers-modal-content">
             <WelcomeMessage fileName={fileName} />
             <ul>
-                {questions &&
-                    questions.map(({ prompt, error, answer = '' }, index) => {
-                        const hasError = !!error;
-                        return (
-                            <li key={index}>
-                                <Question currentUser={currentUser} prompt={prompt} />
-                                {hasError ? <InlineError /> : <Answer answer={answer} isLoading={isLoading} />}
-                            </li>
-                        );
-                    })}
+                {questions?.map(({ prompt, error, answer = '' }, index) => {
```

Comment:
nit: sorry didn't see this before but `questions` is a required prop and it's defaulted to an empty array so the `?.` check isn't necessary